src/naturallanguage.cs

```diff
@@ -0,0 +1,246 @@
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+using System;
+using System.ComponentModel;
+using Foundation;
+using ObjCRuntime;
+
+namespace NaturalLanguage {
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[DisableDefaultCtor] // designated
+	[BaseType (typeof(NSObject))]
+	interface NLLanguageRecognizer {
+
+		[DesignatedInitializer]
+		[Export ("init")]
+		IntPtr Constructor ();
+
+		[Static]
+		[Internal]
+		[Export ("dominantLanguageForString:")]
+		[return: NullAllowed]
+		NSString _GetDominantLanguage (IntPtr @string);
+
+		[Export ("processString:")]
+		void Process (string @string);
+
+		[Export ("reset")]
+		void Reset ();
+
+		[Internal]
+		[NullAllowed, Export ("dominantLanguage")]
+		NSString _DominantLanguage { get; }
+
+		[Wrap ("NLLanguageExtensions.GetValue (_DominantLanguage)")]
+		NLLanguage DominantLanguage { get; }
+
+		// left in case the user does not want to get a c# dict
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("languageHypothesesWithMaximum:")]
+		NSDictionary<NSString, NSNumber> GetNativeLanguageHypotheses (nuint maxHypotheses);
+
+		// left in case the user does not want to get a c# dict
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("languageHints", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSNumber> NativeLanguageHints { get; set; }
+
+		[Internal]
+		[Export ("languageConstraints", ArgumentSemantic.Copy)]
+		NSString[] _LanguageConstraints { get; set; }
+
+		NLLanguage[] LanguageConstraints {
+			[Wrap ("Array.ConvertAll (_LanguageConstraints, e => NLLanguageExtensions.GetValue (e))")]
+			get;
+			[Wrap ("_LanguageConstraints = Array.ConvertAll (value, e => NLLanguageExtensions.GetConstant (e))")]
+			set;
+		}
+	}
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[BaseType (typeof(NSObject))]
+	interface NLModelConfiguration : NSCopying, NSSecureCoding {
+		[Export ("type")]
+		NLModelType Type { get; }
+
+		[Internal]
+		[NullAllowed, Export ("language")]
+		NSString _Language { get; }
+
+		NLLanguage Language { [Wrap ("(_Language.Handle != IntPtr.Zero)? NLLanguageExtensions.GetValue (_Language) : NLLanguage.Undetermined")] get; }
```

Comment:
what if `_Language` is `null` ? in that case an NRE will happen
there's normally no instance with an `Handle` set of `IntPtr.Zero` since using it would generally crash